### PR TITLE
[RLlib] Make sure we step() after adding init_obs.

### PR DESCRIPTION
**Pull request by gjoliver**
## Why are these changes needed?

Bug fix. Always step() after add_init_obs() so we start at 0 env step.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [*] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by kouroshHakha — COMMENTED** [review]
Thanks a ton man. This makes a lot of sense. Can we create a unittest that captures the previous issue? This should be part of env_runner_v2's unittest.

File: rllib/evaluation/tests/test_env_runner_v2.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+from ray.tune import register_env
+
+
+register_env("basic_multiagent", lambda _: BasicMultiAgent(2))
+
+
+class TestEnvRunnerV2(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        ray.init(num_cpus=1, local_mode=True)
```
Comment (kouroshHakha) [inline rllib/evaluation/tests/test_env_runner_v2.py:18]: Don't use local_model and num_cpus once the test is finalized.

**Review by kouroshHakha — APPROVED** [review]
Just a small change requested. Approved o.w. thanks.

File: rllib/evaluation/tests/test_env_runner_v2.py (hunk 1 above)
Comment (gjoliver) [reply rllib/evaluation/tests/test_env_runner_v2.py]: oops, done

**Review by kouroshHakha — APPROVED** [review]

**Pull request merged**